From 7c6a52525ae993b24f9744df2dace5833002fd37 Mon Sep 17 00:00:00 2001
From: Alex Cruz <du.alexcruz@gmail.com>
Date: Wed, 21 Jun 2017 22:05:39 +0200
Subject: [PATCH 023/296] Fix NPE with void
 android.view.VelocityTracker.clear()

 E AndroidRuntime: FATAL EXCEPTION: main
 E AndroidRuntime: Process: com.android.settings, PID: 30816
 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method
 'void android.view.VelocityTracker.clear()' on a null object reference
---
 core/java/android/widget/ScrollView.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/java/android/widget/ScrollView.java b/core/java/android/widget/ScrollView.java
index 97d32f1266a..799339308f3 100644
--- a/core/java/android/widget/ScrollView.java
+++ b/core/java/android/widget/ScrollView.java
@@ -692,7 +692,9 @@ public class ScrollView extends FrameLayout {
                     if (overScrollBy(0, deltaY, 0, mScrollY, 0, range, 0, mOverscrollDistance, true)
                             && !hasNestedScrollingParent()) {
                         // Break our velocity if we hit a scroll barrier.
-                        mVelocityTracker.clear();
+                        if (mVelocityTracker != null) {
+                            mVelocityTracker.clear();
+                        }
                     }
 
                     final int scrolledDeltaY = mScrollY - oldY;
-- 
2.11.0

